manifests/init.pp

```diff
@@ -36,38 +36,28 @@
         message  => "The 'reports' parameter is being deprecated in favor of having the module automatically add the 'splunk_hec' setting to puppet.conf. You can enable this behavior by setting 'reports' to '', the empty string, but please keep in mind that the 'reports' parameter will be removed in a future release.",
         loglevel =>  'warning',
       }
-      $reports_setting = $reports
+
+      pe_ini_setting {'enable splunk_hec':
+        ensure  => present,
+        path    => '/etc/puppetlabs/puppet/puppet.conf',
+        section => 'master',
+        setting => 'reports',
+        value   => $reports,
+        notify  => Service['pe-puppetserver'],
+      }
     } else {
-      # $reports == '' so we dynamically calculate the reports
-      # setting. Idea here is that if the user already included
-      # the 'splunk_hec' report processor, we return the setting as-is.
-      # Otherwise, we return the setting _with_ the report processor included.
-      #
-      # Note: much of this code was inspired by https://github.com/puppetlabs/puppet/blob/6.16.0/lib/puppet/transaction/report.rb.
-      # Also we use inline_template because $settings::reports != Puppet[:reports] and we want Puppet[:reports] since that's
-      # what the report processor code (transaction/report.rb) uses.
-      $raw_reports_setting = inline_template('<%= Puppet[:reports] %>')
-      if $raw_reports_setting == 'none' {
-        $reports_setting = 'splunk_hec'
-      } else {
-        $reports_array = split(regsubst($raw_reports_setting, /(^\s+)|(\s+$)/, '', 'G'), /\s*,\s*/)
-        if 'splunk_hec' in $reports_array {
-          # Use the raw setting so that Puppet won't mark it as changed
-          $reports_setting = $raw_reports_setting
-        } else {
-          $reports_setting = join($reports_array + ['splunk_hec'], ', ')
-        }
+      # The subsetting resource automatically adds the 'splunk_hec' report
+      # processor to the reports setting if it hasn't yet been added there.
+      pe_ini_subsetting { 'enable splunk_hec':
```

Comment:
We should probably have an acceptance test for this to make sure it works.

Reply:
I was asked some of Principal Engineers to please not use the pe_init_settings module in modules published to the forge. They basically want that module as a private dependency for PE internal stuff only. That's one of the reasons why if you try to use this module against Pupperware, this resource will fail. It's not included in that image build because they are confident no one like us should be using it and have no intention of adding it into the container. I know we aren't using that container, but it's an indication of their attitude regarding this resource.

Reply:
@RandomNoun7 Yes, agreed. However, the module already used PE module resources (e.g. before this, it was `pe_ini_setting`). I left it as-is b/c I didn't want to break existing Splunk-hec users by introducing an extra dependency to the `inifile` module.

We should definitely remove the PE-specific resources in the future though.

Reply:
I forgot this was splunc_hec that was already using pe_init_setting. Yeah I can agree suddenly removing it in a change like this isn't necessary.

Reply:
@ekinanp working on adapting Bills acceptance framework to use a container published by splunk. Will look into adding acceptance tests once that's up and working.

Reply:
Cool, thanks both.